### PR TITLE
(Hopefully) Lessens preferences and manifest injection lag

### PR DESCRIPTION
**Pull request by silicons**
Makes it reuse one mob instead of making and deleting one each time it generates a preview.
Tested to work with one person, no idea how it'll behave on live, hope this can get a testmerge.

### CODE REVIEW

File: code/__HELPERS/icons.dm [hunk 1]
```diff
@@ -925,6 +925,27 @@ GLOBAL_LIST_EMPTY(friendly_animal_types)
...
+	if(!GLOB.human_dummy_list[slotkey])
+		var/mob/living/carbon/human/dummy/D = new
+		GLOB.human_dummy_list[slotkey] = D
+		D.in_use = TRUE
+		return D
+	else
+		var/mob/living/carbon/human/dummy/D = GLOB.human_dummy_list[slotkey]
+		UNTIL(!D.in_use)
+		D.in_use = TRUE
+		return D
```
Comment (Cyberboss) [inline code/__HELPERS/icons.dm]: Move these two lines out of each if

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/__HELPERS/icons.dm [hunk 2]
```diff
@@ -925,11 +925,33 @@ GLOBAL_LIST_EMPTY(friendly_animal_types)
 		return J
 	return 0
 
+//Inefficient pooling/caching way.
+GLOBAL_LIST_INIT(human_dummy_list, list())	//Dummy mob = In use (TRUE/FALSE)
```
Comment (Cyberboss) [inline code/__HELPERS/icons.dm]: GLOBAL_LIST_EMPTY

File: code/__HELPERS/icons.dm [hunk 3]
```diff
@@ -925,11 +925,33 @@ GLOBAL_LIST_EMPTY(friendly_animal_types)
 		return J
 	return 0
 
+//Inefficient pooling/caching way.
+GLOBAL_LIST_INIT(human_dummy_list, list())	//Dummy mob = In use (TRUE/FALSE)
+
+/proc/generate_or_wait_for_human_dummy(slotkey)
```
Comment (Cyberboss) [inline code/__HELPERS/icons.dm]: Why not put all this crap in dummy.dm, then you wouldn't have to move the UNTIL macro

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/dummy.dm [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	var/in_use = FALSE
+
+INITIALIZE_IMMEDIATE(/mob/living/carbon/human/dummy)
+
+/mob/living/carbon/human/dummy/Life()
+	return
+
+/mob/living/carbon/human/dummy/proc/wipe_state()
+	QDEL_LIST(contents)
+
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/dummy.dm]: probably should cut_overlays(TRUE) here as well

File: code/modules/mob/living/carbon/human/dummy.dm [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+GLOBAL_LIST_EMPTY(human_dummy_list)
+
+/proc/generate_or_wait_for_human_dummy(slotkey)
+	if(!slotkey)
+		return new /mob/living/carbon/human/dummy
+	var/mob/living/carbon/human/dummy/D = GLOB.human_dummy_list[slotkey]
+	if(!D)
+		D = new
+		GLOB.human_dummy_list[slotkey] = D
+	UNTIL(!D.in_use)
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/dummy.dm]: what if D is QDELETED during the UNTIL?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/dummy.dm [hunk 6]
```diff
@@ -0,0 +1,39 @@
+
+/mob/living/carbon/human/dummy
+	real_name = "Test Dummy"
+	status_flags = GODMODE|CANPUSH
+	var/in_use = FALSE
+
+INITIALIZE_IMMEDIATE(/mob/living/carbon/human/dummy)
+
+/mob/living/carbon/human/dummy/Life()
```
Comment (Cyberboss) [inline code/modules/mob/living/carbon/human/dummy.dm]: Add a Destroy() above here and set in_use to FALSE in it

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

File: code/modules/mob/living/carbon/human/dummy.dm [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+GLOBAL_LIST_EMPTY(human_dummy_list)
+
+/proc/generate_or_wait_for_human_dummy(slotkey)
+	if(!slotkey)
+		return new /mob/living/carbon/human/dummy
+	var/mob/living/carbon/human/dummy/D = GLOB.human_dummy_list[slotkey]
+	if(istype(D))
+		UNTIL(!D.in_use)
+	else
+		pass()
```
Comment (MrStonedOne) [inline code/modules/mob/living/carbon/human/dummy.dm]: WHAT IS THAT

WHAT. THE. FUCK. IS. THAT?

**Pull request merged**